### DISCUSSION TITLE
umi搭配ant-design-mobile 使用 yarn、pnpm 打包出来有不同的className

### DISCUSSION THREAD

**Discussion by shifenhutu** [umijs/umi#8150] [Q&A] [answered]
重现代码：
```jsx
import {Modal} from "antd-mobile";


   <a  onClick={()=>{
        Modal.confirm({
          content: '123123123123',
        });
      }}>
        确认
      </a>
```

## 不同系统、包管理工具，启动、build出来的className：

linux + yarn  `adm-center-popup adm-modal`
win10 + yarn `adm-center-popup adm-modal ` 
mac + yarn  `adm-center-popup adm-modal`

win10 pnpm `adm-modal `
mac + pnpm   `adm-modal `

win10+ npm   `adm-modal `
mac + npm `adm-center-popup adm-modal`

简单得出结果， yarn 多出来一个  `adm-center-popup`,  mac上 npm也会多出来 `adm-center-popup`,  pnpm似乎能保持一致性

# 线上的
<img width="835" alt="image" src="https://user-images.githubusercontent.com/104620424/172768764-430395f1-331c-42fa-9c02-7d337b414f68.png">

# 本地的
<img width="746" alt="image" src="https://user-images.githubusercontent.com/104620424/172768577-9659e9ed-ada0-4727-aa32-89c43b75649f.png">

-----

也就是说。到了线上， modal 代码 被 popup给污染了，但是这怎么会呢？ 太奇怪了

有大佬知道如何排查问题吗？

谢谢

```
  "dependencies": {
    "@umijs/plugin-antd-mobile": "^1.2.0",
    "ahooks": "^3.3.0",
    "antd": "^4.20.7",
    "antd-mobile": "^5.11.0",
    "antd-mobile-icons": "^0.2.2",
    "classnames": "^2.3.1",
    "lib-flexible": "^0.3.2",
    "react": "17.x",
    "react-dom": "17.x",
    "style-resources-loader": "^1.4.1",
    "umi": "^3.5.23",
    "umi-request": "^1.4.0",
   ...
  },
```

**Comment by awmleer** [comment, marked as answer]
`adm-center-popup` 是 CenterPopup 的类名，而 CenterPopup 组件是在 `antd-mobile` 5.14.0 版本新增的，在这个版本中，Modal 也的底层实现也改成基于 CenterPopup 实现了，所以看上去的最终效果是会多一个 `adm-center-popup` 的类名。详见[发布日志](https://github.com/ant-design/ant-design-mobile/releases/tag/v5.14.0)。

所以推测看下来估计是因为 lockfile 导致的本地和线上构建环境的 `antd-mobile` 版本不一致导致的。

**Reply by shifenhutu** [reply]
谢谢，经过我的测试， ， 问题可能来自于 我的umi配置里的：  

```
  extraBabelPlugins: [
    [
      'import',
      {
        libraryName: 'antd-mobile',
        libraryDirectory: 'es/components',
        style: false,
      },
      'antd-mobile',
    ],
  ],
```


我把这个配置换成：

```
 antd: { mobile: false }, 
```

似乎样式就没有bug了
